<?xml version="1.0"?>
<launch>
  <!--node pkg="nodelet" type="nodelet" name="pcl_manager"  args="manager" output="screen"/-->

  <arg name="start_manager" default="true"/>
	<arg name="nodelet_manager" default="extract_planes_manager"/>
	<node if="$(arg start_manager)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"/>

   <!-- Run a VoxelGrid filter to clean NaNs and downsample the data -->
  <!--node pkg="nodelet" type="nodelet" name="voxel_grid2" args="load pcl/VoxelGrid cam3d_nodelet_manager" output="screen">
    <remap from="~input" to="/aggregate_point_map/pc_aligned"/>
    <rosparam>
      filter_field_name: z
      filter_limit_min: 0.2
      filter_limit_max: 3
      filter_limit_negative: False
      leaf_size: 0.02
      input_frame: "/map"
      output_frame: "/map"
    </rosparam>
  </node-->

  <node pkg="nodelet" type="nodelet" name="plane_extraction" args="load cob_3d_mapping_features/PlaneExtractionNodelet $(arg nodelet_manager)" output="screen">
  	<remap from="point_cloud2" to="registration/pc_aligned"/>
  	<remap from="chull" to="plane_extraction/chull"/>
  	<remap from="plane_marker" to="plane_extraction/plane_marker"/>
  	<remap from="object_cluster" to="plane_extraction/object_cluster"/>
  	<remap from="polygons" to="plane_extraction/polygons"/>
  	<remap from="polygon_array" to="plane_extraction/polygon_array"/>
  	<remap from="shape_array" to="plane_extraction/shape_array"/>
  	<!--remap from="point_cloud2" to="/aggregate_point_map/map"/-->
  	<!--remap from="point_cloud2" to="/voxel_grid2/output"/-->
  	<param name="file_path" type="string" value="/home/goa/tmp"/>
  	<param name="save_to_file" type="bool" value="false"/>
  	<param name="plane_constraint" type="int" value="0"/>
  	<param name="passthrough_max_z" type="double" value="3.0"/>
  </node>

</launch>
